Improve setup_lcg.sh and update minimum ROOT version to 6.28

jeanyvesb9 opened the pull request:
This PR adds support for StatAnalysis and multiple OS version in the `setup_lcg.sh` script.
It also updates the minimum ROOT version requirement to 6.28 (already up-to-date in the README)

longjon929 commented:
Hi @jeanyvesb9, thanks.  Two comments.  One is that LCG104 uses a newer ROOT release.  I would prefer to separate any updates to ROOT so that we can update the recommendations in various places and check for differences.  Second, this didn't work on lxplus out of the box, the arch string seems to be wrong:
```
source setup_lcg.sh 
Error (views) : views release not found
Error: incorrect x86_64-centos7-gcc13-opt <arch-os-complier> for LCG_104; available: 
```

jeanyvesb9 commented:
Hi @longjon929. I fixed the centos7 problem (gcc12 was required instead of gcc13; changed it for both centos7 and el9 for consistency). But regarding the Root version, the recommended StatAnalysis branch also has Root 6.28.04, the same as LCG_104. That's why I updated it to that newer version.

longjon929 commented:
Hi, right it's already in the stat release.  I was just suggesting this as there are many places to update the release that we could do comprehensively separate from this, eg could you please update this line in the README https://github.com/histfitter/histfitter/blob/master/README.md?plain=1#L36

jeanyvesb9 commented:
Done! I also updated the root version in the Dockerfile, and added a check for the minor revision of the minimum version in `histfitter_env_setup.sh`. But I would maintain the minimum requirement in the `histfitter_env_setup.sh` script to 6.28/00, since it would also work fine with that revision.
Aside from this "special" situation where we're addressing a version mismatch, I completely agree with you in changing root versions in dedicated commits/histfitter revisions.
